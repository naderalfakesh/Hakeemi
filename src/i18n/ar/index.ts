import common from './common.json';
import home from './home.json';
import signIn from './signIn.json';
import signup from './signup.json';
import welcome from './welcome.json';

export default {
  common,
  home,
  signIn,
  signup,
  welcome,
};
