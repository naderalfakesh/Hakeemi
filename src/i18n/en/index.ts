import common from './common.json';
import home from './home.json';
import login from './login.json';
import signup from './signup.json';
import welcome from './welcome.json';

export default {
  common,
  home,
  login,
  signup,
  welcome,
};
