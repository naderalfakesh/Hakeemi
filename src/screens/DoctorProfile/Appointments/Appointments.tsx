import React from 'react';
import { View } from 'react-native';
import Button from '../../../components/Button';
import Text from '../../../components/Text';
import styles from './styles';
import { useTranslation } from 'react-i18next';

interface Props {
  title: string;
  availableDates: Array<Date>;
  onPress: (date: Date) => void;
}
const Appointments = ({ title, availableDates, onPress }: Props) => {
  const { t } = useTranslation('doctorProfile');
  return (
    <View style={styles.container}>
      <Text color="black" size="head-24" style={styles.title}>
        {title}
      </Text>
      <View style={styles.listContainer}>
        {availableDates.map((date, index) => (
          <Button
            style={styles.button}
            key={index}
            onPress={() => onPress(date)}
            theme="tertiary">
            {t('dateFormat', { date })}
          </Button>
        ))}
      </View>
    </View>
  );
};

export default Appointments;
