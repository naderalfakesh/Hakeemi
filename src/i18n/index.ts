import i18n from './i18n';

export * from './i18n';

export default i18n;
