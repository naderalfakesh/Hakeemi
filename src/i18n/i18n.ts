import i18next from 'i18next';
import en from './en';
import ar from './ar';

const i18n = i18next.createInstance();
i18n.init({
  fallbackLng: 'en',
  lng: 'en',
  supportedLngs: ['en', 'ar'],
  defaultNS: 'common',
  resources: {
    en,
    ar,
  },
  react: {
    useSuspense: false,
  },
});

export default i18n;
