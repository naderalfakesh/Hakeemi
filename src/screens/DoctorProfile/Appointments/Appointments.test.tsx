import React, { FC } from 'react';
import { render } from '@testing-library/react-native';
import Appointments from './Appointments';
import { I18nextProvider } from 'react-i18next';
import i18n from '../../../i18n';

const wrapper: FC = ({ children }) => {
  return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;
};
describe('Button component test', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <Appointments
        title="Title"
        availableDates={[
          new Date(2021, 5, 5, 10, 45, 0),
          new Date(2021, 5, 6, 15, 18, 0),
        ]}
        onPress={() => {}}
      />,
      { wrapper },
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
