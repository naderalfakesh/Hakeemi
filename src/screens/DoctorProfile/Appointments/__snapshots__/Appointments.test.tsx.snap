// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 32,
      "paddingHorizontal": 32,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 24,
        "fontWeight": "400",
        "includeFontPadding": false,
        "lineHeight": 30,
        "marginBottom": 10,
        "textAlign": undefined,
      }
    }
    testID="Text"
  >
    Title
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#edf3f7",
          },
          Object {
            "padding": 16,
          },
          false,
          false,
          Object {
            "marginBottom": 10,
            "marginRight": 10,
          },
        ]
      }
      testID="Button"
    >
      <Text
        style={
          Object {
            "color": "#00bbd4",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 20,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        Sat, 5-Jun
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#edf3f7",
          },
          Object {
            "padding": 16,
          },
          false,
          false,
          Object {
            "marginBottom": 10,
            "marginRight": 10,
          },
        ]
      }
      testID="Button"
    >
      <Text
        style={
          Object {
            "color": "#00bbd4",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 20,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        Sun, 6-Jun
      </Text>
    </View>
  </View>
</View>
`;
